cpp/cmake/sanitizers.cmake

```diff
@@ -204,6 +222,48 @@ function(cpp_cc_configure_sanitizers)
   endforeach()
 endfunction(cpp_cc_configure_sanitizers)
 
+# Helper function strips away Python shims on macOS, so we can launch tests using the actual Python
+# binary. Without this, preloading the sanitizer runtimes does not work on macOS. See
+# https://jonasdevlieghere.com/sanitizing-python-modules/ and
+# https://tobywf.com/2021/02/python-ext-asan/ for more information.
```

Comment:
interesting!

Reply:
@pramodk thinking of SKIP_EMBEDDED_PYTHON_TEST? :)

Reply:
while working on wheels with the apple security settings etc, I am glad that I/we didn't come across this!